glean-core/python/glean/glean.py

```diff
@@ -213,6 +209,37 @@ def initialize():
 
         cls._init_finished = True
 
+    @classmethod
+    def _initialize_with_tempdir_for_testing(
+        cls,
+        application_id: str,
+        application_version: str,
+        upload_enabled: bool,
+        configuration: Optional[Configuration] = None,
+        data_dir: Optional[Path] = None,
+        application_build_id: Optional[str] = None,
+    ) -> None:
+        """
+        Initialize Glean to use a temporary data directory. Use for internal
+        unit testing only.
+
+        The temporary directory will be destroyed when Glean is initialized
+        again or at process shutdown.
+        """
+        if data_dir is None:
+            actual_data_dir = Path(tempfile.TemporaryDirectory().name)
+        else:
+            actual_data_dir = data_dir
```

Comment:
Is there any reason to keep this behavior and not just always use a temp directory when this function is used?

Reply:
Yep -- for example: https://github.com/mozilla/glean/blob/main/glean-core/python/tests/test_glean.py#L156

Reply:
But ... that's calling `Glean.initialize` already, which handles the data_dir.
The test-only function here (`_initialize_with_tempdir_for_testing`) shouldn't need it?

Reply:
Ah, I see your point, now.  Makes sense.  I'll simplify this (and update the tests).